src/Service/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -506,5 +482,67 @@ public void ProcessOdataClause(FilterClause odataClause)
         {
             return filterClause.Expression.Accept<string>(visitor);
         }
+
+        /// <summary>
+        /// Gets the value of the parameter cast as the system type
+        /// </summary>
+        /// <param name="param">Parameter value as a string</param>
+        /// <param name="paramName">Parameter name</param>
+        /// <param name="systemType">System type to which the parameter value is parsed to</param>
+        /// <returns>The parameter value parsed to the specified system type</returns>
+        /// <exception cref="DataApiBuilderException">Throws a DataApiBuilderException when the conversion of parameter value to the specified system type fails. The error message returned will be different in development
+        /// and production modes. In production mode, the error message returned will be generic so as to not reveal information about the database object backing the entity</exception>
+        protected object GetParamAsSystemType(string param, string paramName, Type systemType)
+        {
+            try
+            {
+                return ParseParamAsSystemType(param, systemType);
+            }
+            catch (Exception e)
+            {
+                if (e is FormatException ||
+                    e is ArgumentNullException ||
+                    e is OverflowException)
+                {
+                    string errorMessage;
+                    if (MetadataProvider.IsDevelopmentMode())
+                    {
+                        if (MetadataProvider.EntityToDatabaseObject[EntityName].SourceType is SourceType.StoredProcedure)
+                        {
+                            errorMessage = $@"Parameter ""{param}"" cannot be resolved as stored procedure parameter ""{paramName}"" " +
```

Comment:
Does it matter whether we include the database object type in the error message here? does that change how a developer acts on the message? (I don't have a yes or no here, just curious since if it does not matter, you could simplify this code by not having all the extra conditionals.

Reply:
The developers action would be to use the right value for the parameters in the requests irresective of the database object type. 

My thought was that displaying details such as whether the entity is a stored procedure/table/view along with the parameter/column would be helpful (essentially, more the details, more helpful it would be) for them to determine the right type to use in the requests.